### PR TITLE
promisify and lipstickify

### PR DESCRIPTION
**Pull request by SamVerschueren**
juicycouturesheep

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -17,31 +17,32 @@ $ npm install --save jsftp-mkdirp
...
 });
 ```
 
 
 ## API
 
-### JSFtp.mkdirp(path, callback)
+### JSFtp.mkdirp(path)
+
+Returns a Promise.
```
Comment (SamVerschueren) [inline readme.md]: Feedback?

File: test.js [hunk 2]
```diff
@@ -23,32 +23,34 @@ before(function (cb) {
...
+	setTimeout(() => {
 		ftp = new JSFtp({
 			host: 'localhost',
 			port: 3334,
 			user: 'test',
 			pass: 'test'
 		});
 
-		cb();
+		t.end();
```
Comment (SamVerschueren) [inline test.js]: Somehow this ugly hack is necessary for travis. WTF!?

File: readme.md (hunk 1 above)
Comment (kevva) [inline readme.md]: `promise`?

File: index.js [hunk 3]
```diff
@@ -1,54 +1,51 @@
...
 	}
 
-	var mkdir = function (dir) {
+	const mkdir = dir => {
 		dir = slash(dir);
 
-		this.raw.mkd(dir, function (err) {
-			if (err && err.code === 550) {
+		return pify(this.raw.mkd.bind(this.raw))(dir)
+			.then(() => {
```
Comment (kevva) [inline index.js]: `.then(() => dirs.length > 1 && mkdir(dirs.pop()))` would work.

Comment (SamVerschueren) [inline index.js]: :+1:

Comment (kevva) [inline index.js]: `dirs.length > 0` that is, hah.

File: readme.md (hunk 1 above)
Comment (sindresorhus) [inline readme.md]: @kevva `promise` looks better and what I've been using, but `Promise` is more correct as it's a name.

File: test.js (hunk 2 above)
Comment (sindresorhus) [inline test.js]: You need that since it's a `cb` test. Use async function and normal test instead and https://github.com/sindresorhus/delay

Comment (SamVerschueren) [inline test.js]: Ow yes, delay would make it much cleaner, good catch! :)

**Comment by SamVerschueren** [comment]
processed feedback

**Comment by sindresorhus** [comment]
Yay! This must have been a ton of work. Really appreciate it.

# Promisify all the things

**Comment by ArtskydJ** [comment]
I admire the push to use `Promise`s, but I wonder if it would be better to have this module be callback based. This module is an extension to `jsftp`, which is totally callback based, but `ftp.mkdirp` exposes a very different API from all the other `ftp` methods.

Just a thought...

**Comment by sindresorhus** [comment]
@ArtskydJ Nah. We can't do anything about `jsftp`, but at least we can make this extension good.

**Pull request merged**